Windows Support

Tool doesn't appear to run on windows. Ran fine on my linux PC though.

Run the update still getting issues, takes about 15 - 20 minutes to run.

```
C:\js_source\iron-dropdown>npm run update-types

> @polymer/iron-dropdown@ update-types C:\js_source\iron-dropdown
> bower install && gen-typescript-declarations --deleteExisting --outDir .

Loading config from "gen-tsd.json".

<--- Last few GCs --->

[17992:000002812403BE70]  2485898 ms: Mark-sweep 4099.3 (4312.9) -> 4099.3 (4312.9) MB, 9016.7 / 0.0 ms  last resort GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 000000A33AB25EE1 <JSObject>
    2: /* anonymous */(aka /* anonymous */) [C:\Users\jhome\AppData\Roaming\npm\node_modules\@polymer\gen-typescript-declarations\node_modules\babylon\lib\index.js:839] [bytecode=00000063B26FAC81 offset=59](this=00000056BA102311 <undefined>,key=000003B1ACA38D59 <String[13]: containsOctal>)
    3: arguments adaptor frame: 3->1
    4: forEach(this=00000114F427E3F1 <JSArray[45]>)
    5: clone [C:...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node_module_register
 2: v8::internal::FatalProcessOutOfMemory
 3: v8::internal::FatalProcessOutOfMemory
 4: v8::internal::Factory::NewStruct
 5: std::vector<v8::internal::compiler::MoveOperands * __ptr64,v8::internal::ZoneAllocator<v8::internal::compiler::MoveOperands * __ptr64> >::_Reallocate
 6: std::vector<v8::internal::compiler::MoveOperands * __ptr64,v8::internal::ZoneAllocator<v8::internal::compiler::MoveOperands * __ptr64> >::_Reallocate
 7: std::vector<v8::internal::compiler::MoveOperands * __ptr64,v8::internal::ZoneAllocator<v8::internal::compiler::MoveOperands * __ptr64> >::_Reallocate
 8: std::vector<v8::internal::compiler::MoveOperands * __ptr64,v8::internal::ZoneAllocator<v8::internal::compiler::MoveOperands * __ptr64> >::_Reallocate
 9: std::vector<v8::internal::compiler::MoveOperands * __ptr64,v8::internal::ZoneAllocator<v8::internal::compiler::MoveOperands * __ptr64> >::_Reallocate
10: std::vector<v8::internal::compiler::MoveOperands * __ptr64,v8::internal::ZoneAllocator<v8::internal::compiler::MoveOperands * __ptr64> >::_Reallocate
11: std::vector<v8::internal::compiler::MoveOperands * __ptr64,v8::internal::ZoneAllocator<v8::internal::compiler::MoveOperands * __ptr64> >::_Reallocate
12: 00000268963047A1
npm ERR! code ELIFECYCLE
npm ERR! errno 3
npm ERR! @polymer/iron-dropdown@ update-types: `bower install && gen-typescript-declarations --deleteExisting --outDir .`
npm ERR! Exit status 3
npm ERR!
npm ERR! Failed at the @polymer/iron-dropdown@ update-types script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\jhome\AppData\Roaming\npm-cache\_logs\2018-03-18T16_07_32_227Z-debug.log
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,7 +4,8 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
-<!-- ## [Unreleased] -->
+## [Unreleased]
+- Fix bad file path handling which broke Windows support.
 
 ## [1.2.1] - 2018-03-09
 - Fix bug where if a package name was a prefix of one of its dependencies (e.g.

--- a/package.json
+++ b/package.json
@@ -22,7 +22,8 @@
     "fs-extra": "^5.0.0",
     "glob": "^7.1.2",
     "minimatch": "^3.0.4",
-    "polymer-analyzer": "=3.0.0-pre.14"
+    "polymer-analyzer": "=3.0.0-pre.14",
+    "vscode-uri": "^1.0.3"
   },
   "devDependencies": {
     "@types/chai": "^4.0.4",

--- a/src/gen-ts.ts
+++ b/src/gen-ts.ts
@@ -13,7 +13,7 @@ import * as minimatch from 'minimatch';
 import * as path from 'path';
 import * as analyzer from 'polymer-analyzer';
 import {Function as AnalyzerFunction} from 'polymer-analyzer/lib/javascript/function';
-import * as url from 'url';
+import Uri from 'vscode-uri';
 
 import {closureParamToTypeScript, closureTypeToTypeScript} from './closure-types';
 import * as ts from './ts-ast';
@@ -183,11 +183,11 @@ function analyzerToAst(
  */
 function analyzerUrlToRelativePath(
     analyzerUrl: string, rootDir: string): string|undefined {
-  const parsed = url.parse(analyzerUrl);
-  if (parsed.protocol !== 'file:' || parsed.host || !parsed.path) {
+  const parsed = Uri.parse(analyzerUrl);
+  if (parsed.scheme !== 'file' || parsed.authority || !parsed.fsPath) {
     return undefined;
   }
-  return path.relative(rootDir, parsed.path);
+  return path.relative(rootDir, parsed.fsPath);
 }
 
 /**

